File: striver/S2L2/rbubble.py
size = int(input())
arr = [0]*size
for i in range(0,size):
    arr[i]=int(input())
def rbs(arr,size):
    if size==1 :
        return
    for j in range(0,size-1):
        if arr[j]>=arr[j+1]:
            temp = arr[j]
            arr[j] = arr[j+1]
            arr[j+1] = temp
    rbs(arr,size-1)
rbs(arr,size)
for i in range(0,size):
        print(arr[i],end=" ")

File: striver/S2L2/rbubble.java
import java.util.Scanner;

public class rbubble{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        for(int i=0;i<n;i++)
            arr[i] = sc.nextInt();
        rbs(arr,n);
        for(int i=0;i<n;i++)
            System.out.print(arr[i]+" ");
    }
    public static void rbs(int[] arr,int n){
        if(n==1)return;
        for(int j=0;j<=n-2;j++){
            if(arr[j]>=arr[j+1]){
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
        
        rbs(arr,n-1);
    }
}
